code-challenges/regexp-part2/challenges-11.test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named validatePin that uses a regular expression pattern to validate a PIN.
+
+If the PIN is four numerical digits long, return true. Otherwise, return false.
+------------------------------------------------------------------------------------------------ */
+
+const validatePin = (pin) => {
+  // Solution code here...
+  let validator = /^[0-9]{4}$/
+  return validator.test(pin) ? true : false;
+
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named validateEmail that takes in an email address and validates it based
+on several rules:
+  - one word, or two words separated by a period, before the @ symbol
+  - can contain numbers
+  - can have any of the following top-level domains: .net, .com, or .org
+  - no other special characters
+  - no subdomains, ports, etc: must be of the form name@place.com, not name@sub.place.com:3000
+
+Return either true or false.
+
+Note: if you ever need to validate an email using a regex in practice, the Internet has the actual regex you should use. It's many many lines long.
+------------------------------------------------------------------------------------------------ */
+
+const validateEmail = (email) => {
+  // Solution code here...
+  var validemail =/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  return validemail.test(email) ? true :false;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named validatePhoneNumber that accepts a phone number and determines if it is valid.
+
+Acceptable formats include:
+ - (555) 555-5555
+ - (555)555 5555
+ - 555 555-5555
+ - 555-5555555
+ - 555-555 5555
+ - 555-555-5555
+ - 555 555 5555
+ - 555555-5555
+ - 5555555555
+
+Your function should include a single regular expression pattern that matches any of these formats.
+
+Return either true or false.
+------------------------------------------------------------------------------------------------ */
+
+const validatePhoneNumber = (phoneNumber) => {
+  // Solution code here...
+  const regex = /^(1\s|1|)?((\(\d{3}\))|\d{3})(\-|\s)?(\d{3})(\-|\s)?(\d{4})$/;       // I google it 
+  return phoneNumber.match(regex) ? true:false;
```

Comment:
I need to know from where you googled it. Very important to cite sources.

I don't expect you to be able do and advanced expression on your own, but please do give links

Reply:
I am sorry not in purpose, and I already have put it in my code that I am used google to get the answer, wallah it's my first time did that but I learnt, thanks